### PR TITLE
Bug 705767 - Validate credentials during setup

### PR DESCRIPTION
**Pull request by liuche**
Rebased version of pull #143.

### CODE REVIEW

File: src/main/java/org/mozilla/gecko/sync/setup/activities/AccountActivity.java [hunk 1]
```diff
@@ -177,53 +222,96 @@ private boolean validateInputs() {
...
-      runOnUiThread(new Runnable() {
-        @Override
-        public void run() {
-          authFailure();
-        }
-      });
-      return;
+  public void authCallback(final AuthenticationResult result) {
+    displayVerifying(false);
+    if (result == AuthenticationResult.SUCCESS) {
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/setup/activities/AccountActivity.java]: Reverse these two cases, early exit.

File: src/test/java/org/mozilla/android/sync/net/test/TestAccountAuthenticatorStage.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+              try {
+                authStage.authenticateAccount(testCallback, TEST_SERVER, authHeader);
+              } catch (URISyntaxException e) {
+                fail("Malformed URI.");
+              }
+//            }
+//          });
+//          } catch (InnerError e) {
+//            fail("testWaiter inner error.");
+//          }
```
Comment (rnewman) [inline src/test/java/org/mozilla/android/sync/net/test/TestAccountAuthenticatorStage.java]: Why is all this code commented out?

**Comment by rnewman** [comment]
Tidied, rebased, bug number added to commit messages, commit messages adjusted to more closely match their code changes, etc.

**Comment by rnewman** [comment]
Testing this today.

**Comment by rnewman** [comment]
Needs work: apparently incorrectly handles non-hashing usernames.

**Comment by liuche** [comment]
Updated to #202.

**Pull request closed without merging**